doc-content/enterprise-only/project-data/variables-con.adoc

```diff
@@ -19,6 +19,63 @@ If the variable cannot be found directly in the element's context, the immediate
+* `readonly`: Indicates that the variable is for informational purposes only and can be set only once during process instance execution. If the value of a read-only variable is modified at any time, {PRODUCT} generates a `VariableViolationException` error.
+* `restricted`: Indicates a special tag that is used with `VariableGuardProcessEventListener` for granted permission to modify the variable based on the required and the existing role.
++
+`VariableGuardProcessEventListener` is extended from `DefaultProcessEventListener` and supports two different constructors:
++
+[source,java]
+----
+public VariableGuardProcessEventListener(String requiredRole, IdentityProvider identityProvider) {
+    this("restricted", requiredRole, identityProvider);
+}
+
+public VariableGuardProcessEventListener(String tag, String requiredRole, IdentityProvider identityProvider) {
+    this.tag = tag;
+    this.requiredRole = requiredRole;
+    this.identityProvider = identityProvider;
+}
+----
+
++
+Therefore, it is must to add an event listener to the session with allowed role name and identity provider that returns the user role:
++
+[source,java]
+----
+ksession.addEventListener(new VariableGuardProcessEventListener("AdminRole", myIdentityProvider));
+----
++
+`VariableGuardProcessEventListener` verifies if a variable is tagged with a security constraint tag (`restricted`). If the user role is not required, then {PRODUCT} generates a `VariableViolationException` error.
+
+NOTE: The variable tags appearing in the {CENTRAL} UI such as `internal`, `input`, `output`, `business-relevant`, and `tracked` are not supported in {PRODUCT}.
+
+You can add the tag directly in the BPMN process source file as a `customTags` metadata property with the tag value defined in the format `![CDATA[__TAG_NAME__]]`.
+
+For example, the following BPMN process applies the `required` tag to an `approver` process variable:
+
+.Example variable tagged in the BPMN modeler
+image::BPMN2/bpmn-variable-tags-ui.png[Image of variable tags in BPMN modeler]
+
+.Example variable tagged in a BPMN file
+[source,xml]
+----
+<bpmn2:property id="approver" itemSubjectRef="ItemDefinition_9" name="approver">
+  <bpmn2:extensionElements>
+    <tns:metaData name="customTags">
+      <tns:metaValue><![CDATA[required]]></tns:metaValue>
+    </tns:metaData>
+  </bpmn2:extensionElements>
+</bpmn2:property>
+----
+
+You can use multiple tags for a variable where applicable.You can also define custom variable tags in your BPMN files to make variable data available to {PRODUCT} process event listeners. Custom tags do not influence the {PRODUCT} runtime as the standard variable tags do and are for informational purposes only. You define custom variable tags in the same `customTags` metadata property format that you use for standard {PRODUCT} variable tags.
```

Comment:
You can use more than one tag for a variable.

Reply:
Done